### ISSUE TITLE
Create kover-features artifact to invoke the capabilities of Kover via the Java API

### ISSUE THREAD

**Issue by shanshin** [Kotlin/kotlinx-kover#534]
### Motivation
Need to add Kover coverage library support to the Android Gradle Plugin (AGP).

### Current status
AGP only supports working with [JaCoCo](https://www.eclemma.org/jacoco/), it uses the Java calls of JaCoCo methods.

The following JaCoCo API are used:
- [offline instrumentation](https://github.com/jacoco/jacoco/blob/ae26ea0a0fd19a8c00b4866455048ce1f1624761/org.jacoco.core/src/org/jacoco/core/instr/Instrumenter.java#L128) of single class 
- [getting a binary coverage report](https://github.com/jacoco/jacoco/blob/ae26ea0a0fd19a8c00b4866455048ce1f1624761/org.jacoco.agent.rt/src/com/vladium/emma/rt/RT.java#L48)
- generating reports, [html](https://github.com/jacoco/jacoco/blob/971af84dbf29bf7bfecc0279cc6eea99cd7e2f6f/org.jacoco.report/src/org/jacoco/report/html/HTMLFormatter.java#L50), [xml](https://github.com/jacoco/jacoco/blob/971af84dbf29bf7bfecc0279cc6eea99cd7e2f6f/org.jacoco.report/src/org/jacoco/report/xml/XMLFormatter.java#L33), etc

Kover only has an API for getting a binary report in artifact `org.jetbrains.kotlinx:kover-offline-runtime`, a suitable method `kotlinx.kover.offline.runtime.api.KoverRuntime#getReport()` 

### Implementation plan
1. Create published Kover artifact `org.jetbrains.kotlinx:kover-features-jvm`
2. Add Java API for offline instrumentation of single class 
```
package kotlinx.kover.features.jvm

public class KoverFeatures {
    public static OfflineInstrumentator createOfflineInstrumenter() { ... }
}

public interface OfflineInstrumenter {
    public byte[] instrument(InputStream originalClass, String debugName)
}
```
3. Create `kotlinx.kover.features.jvm.KoverLegacyFeatures` and put the methods used by the Kover CLI there
4. Remove the dependency on `intellij-coverage-reporter` from the Kover CLI and add the dependency on `org.jetbrains.kotlinx:kover-features-jvm`, use the methods added in `kotlinx.kover.features.jvm.KoverFeaturesUtils`. Now, `kover-features-jvm` will be the only artifact distributing the Kover Tooling API via Java API.
5. (for the future) Develop an API for generating reports. Analyzing the JaCoCo API, it is obvious that the API for generating reports is not trivial.
It is required to develop an API that implements all the needs of external users (such as AGP), optimized, but as easy to use and non-redundant.

**Comment by shanshin** [comment]
Relates #96

**Comment by shanshin** [comment]
Implemented in `0.7.6`
